src/shared/messaging/port-provider.ts

```diff
@@ -147,8 +147,13 @@ export class PortProvider implements Destroyable {
     const handleRequest = (event: MessageEvent) => {
       const { data, origin, source } = event;
 
+      // Ignore messages that don't look like port requests.
       if (!isMessage(data) || data.type !== 'request') {
-        // If this does not look like a message intended for us, ignore it.
+        return;
+      }
+
+      // Ignore messages where the sender went away before we can send a response.
+      if (!source) {
```

Comment:
Perhaps we could make this as the first check, avoiding `isMessage` to be even called.

Reply:
Done. This is not performance sensitive at all, but it seems reasonable to do the simplest check first.